plugins/hydra_joblib_launcher/hydra_plugins/hydra_joblib_launcher/joblib_launcher.py

```diff
@@ -0,0 +1,136 @@
+        """Joblib Launcher
+
+        Launches parallel jobs using Joblib.Parallel. For details, refer to:
+        https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html
+
+        This plugin is based on the idea and inital implementation of @emilemathieutmp:
+        https://github.com/facebookresearch/hydra/issues/357
+        """
+        self.config: Optional[DictConfig] = None
+        self.config_loader: Optional[ConfigLoader] = None
+        self.task_function: Optional[TaskFunction] = None
+
+        self.joblib = joblib
+
+    def setup(
+        self,
+        config: DictConfig,
+        config_loader: ConfigLoader,
+        task_function: TaskFunction,
+    ) -> None:
+        self.config = config
+        self.config_loader = config_loader
+        self.task_function = task_function
+
+    def launch(self, job_overrides: Sequence[Sequence[str]]) -> Sequence[JobReturn]:
+        """
+        :param job_overrides: a List of List<String>, where each inner list is the arguments for one job run.
+        :return: an array of return values from run_job with indexes corresponding to the input list indexes.
+        """
+        setup_globals()
+        assert self.config is not None
+        assert self.config_loader is not None
+        assert self.task_function is not None
+
+        configure_log(self.config.hydra.hydra_logging, self.config.hydra.verbose)
+        sweep_dir = Path(str(self.config.hydra.sweep.dir))
+        sweep_dir.mkdir(parents=True, exist_ok=True)
+        log.info(
+            "Joblib.Parallel({}) is launching {} jobs".format(
+                ",".join([f"{k}={v}" for k, v in self.joblib.items()]),
+                len(job_overrides),
+            )
+        )
+        log.info("Sweep output dir : {}".format(sweep_dir))
+
+        singleton_state = Singleton.get_state()
+
+        runs = Parallel(**self.joblib)(
+            delayed(dispatch_job)(
+                idx,
+                overrides,
+                self.config_loader,
+                self.config,
+                self.task_function,
+                singleton_state,
+            )
+            for idx, overrides in enumerate(job_overrides)
+        )
+
+        return runs  # type: ignore
+
+
+def dispatch_job(
+    idx: int,
+    overrides: Sequence[str],
+    config_loader: ConfigLoader,
+    config: DictConfig,
+    task_function: TaskFunction,
+    singleton_state: Dict[Any, Any],
+) -> JobReturn:
+    """Calls `run_job` in parallel
+
+    Note that Joblib's default backend runs isolated Python processes, see
+    https://joblib.readthedocs.io/en/latest/parallel.html#shared-memory-semantics
+    """
+    setup_globals()
+    Singleton.set_state(singleton_state)
+
+    log.info("\t#{} : {}".format(idx, " ".join(filter_overrides(overrides))))
+    sweep_config = config_loader.load_sweep_config(config, list(overrides))
+    with open_dict(sweep_config):
+        sweep_config.hydra.job.id = "job_id_for_{}".format(idx)
```

Comment:
this job id is a bit weird.
maybe `${hydra.job.name}_${hydra.job.num}`?

Reply:
Changed